pkg/sdk/testint/users_integration_test.go

```diff
@@ -161,68 +257,690 @@ func TestInt_UserCreate(t *testing.T) {
+			DisplayName:        sdk.String(newValue),
+			FirstName:          sdk.String(newValue),
+			MiddleName:         sdk.String(newValue),
+			LastName:           sdk.String(newValue),
+			Email:              sdk.String(email),
+			MustChangePassword: sdk.Bool(true),
+			Disable:            sdk.Bool(true),
+			DaysToExpiry:       sdk.Int(5),
+			MinsToUnlock:       sdk.Int(15),
+			DefaultWarehouse:   sdk.Pointer(warehouseId),
+			DefaultNamespace:   sdk.Pointer(schemaIdObjectIdentifier),
+			DefaultRole:        sdk.Pointer(roleId),
+			DefaultSecondaryRoles: &sdk.SecondaryRoles{
+				Roles: []sdk.SecondaryRole{{Value: "ALL"}},
+			},
+			MinsToBypassMFA: sdk.Int(30),
+			RSAPublicKey:    sdk.String(key),
+			RSAPublicKey2:   sdk.String(key2),
+			Comment:         sdk.String("some comment"),
+		}}
+
+		err := client.Users.Create(ctx, id, createOpts)
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().User.DropUserFunc(t, id))
+
+		userDetails, err := client.Users.Describe(ctx, id)
+		require.NoError(t, err)
+		assert.Equal(t, id.Name(), userDetails.Name.Value)
+		assert.Equal(t, strings.ToUpper(newValue), userDetails.LoginName.Value)
+		assert.NotEmpty(t, userDetails.Password.Value)
+		assert.Equal(t, newValue, userDetails.MiddleName.Value)
+
+		user, err := client.Users.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		objectAssert.AssertThatObject(t, objectAssert.User(t, user.ID()).
+			HasName(user.Name).
+			HasCreatedOnNotEmpty().
+			// login name is always case-insensitive
+			HasLoginName(strings.ToUpper(newValue)).
+			HasDisplayName(newValue).
+			HasFirstName(newValue).
+			HasLastName(newValue).
+			HasEmail(email).
+			HasMinsToUnlock("14").
+			HasDaysToExpiryNotEmpty().
+			HasComment("some comment").
+			HasDisabled(true).
+			HasMustChangePassword(true).
+			HasSnowflakeLock(false).
+			HasDefaultWarehouse(warehouseId.Name()).
+			HasDefaultNamespaceId(schemaId).
+			HasDefaultRole(roleId.Name()).
+			HasDefaultSecondaryRoles(`["ALL"]`).
+			HasExtAuthnDuo(false).
+			HasExtAuthnUid("").
+			HasMinsToBypassMfa("29").
+			HasOwner(currentRole.Name()).
+			HasLastSuccessLoginEmpty().
+			HasExpiresAtTimeNotEmpty().
+			HasLockedUntilTimeNotEmpty().
+			HasHasPassword(true).
+			HasHasRsaPublicKey(true),
+		)
+	})
+
+	// TODO [SNOW-1348101]: consult this with appropriate team when we have all the problems listed
+	t.Run("create and alter: problems with public key fingerprints", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+
+		createOpts := &sdk.CreateUserOptions{ObjectProperties: &sdk.UserObjectProperties{
+			RSAPublicKey:   sdk.String(key),
+			RSAPublicKeyFp: sdk.String(hash),
+		}}
+
+		err := client.Users.Create(ctx, id, createOpts)
+		require.ErrorContains(t, err, "invalid property 'RSA_PUBLIC_KEY_FP' for 'USER'")
+
+		createOpts = &sdk.CreateUserOptions{ObjectProperties: &sdk.UserObjectProperties{
+			RSAPublicKey2:   sdk.String(key),
+			RSAPublicKey2Fp: sdk.String(hash),
```

Comment:
Isn't FP read-only? I think it means a key fingerprint which is computed from the provided key.

Reply:
Not according to the syntax: https://docs.snowflake.com/en/sql-reference/sql/create-user#syntax